src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestResponse.java

```diff
@@ -1,9 +1,16 @@
 /*
- * Copyright (c) 2022 Snowflake Computing Inc. All rights reserved.
+ * Copyright (c) 2022-2024 Snowflake Computing Inc. All rights reserved.
  */
 
 package net.snowflake.ingest.streaming.internal;
 
+/**
+ * The StreamingIngestResponse class is an abstract class that represents a response from the
+ * Snowflake streaming ingest API. This class provides a common structure for all types of responses
+ * that can be received from the {@link SnowflakeServiceClient}.
+ */
 abstract class StreamingIngestResponse {
   abstract Long getStatusCode();
+
+  abstract String getMessage();
```

Comment:
lets not add this to the interface. I just opened a JIRA to stop this intermediate-string business. We should directly serialize from objects to output json, and deserialize from response bytes to json.

Reply:
The streaming ingest response looks like `{"status_code": 0, "message": "some message"}`. This method is used to log error message from server side.

Reply:
ah ok, i thought this is the whole response as a string